### PR TITLE
[sonic-psud] add a new daemon sonic-psud to platform monitor

### PR DESCRIPTION
**Pull request by kevinwangsk**
* This new daemon will periodcally read PSU presence and status from sysfs and write to DB
* CLI will get the PSU presence and status from DB directly.

Signed-off-by: Kevin Wang <kevinw@mellanox.com>

### CODE REVIEW

File: sonic-psud/scripts/psud [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    # Load platform-specific psuutil class
+    err = load_platform_psuutil()
+    if err != 0:
+        log_error("failed to load psuutil")
+        sys.exit(1)
+
+    state_db = swsscommon.DBConnector(swsscommon.STATE_DB,
+                                     REDIS_HOSTNAME,
+                                     REDIS_PORT,
+                                     REDIS_TIMEOUT_MSECS)
```
Comment (jleveque) [inline sonic-psud/scripts/psud]: Add one more leading space to lines 165-167 so the arguments visually align with the first argument on line 164.

File: sonic-psud/scripts/psud [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    psu_tbl = swsscommon.Table(state_db, "PSU_INFO")
+    chassis_tbl = swsscommon.Table(state_db, "CHASSIS_INFO")
+    psu_num = platform_psuutil.get_num_psus()
```
Comment (jleveque) [inline sonic-psud/scripts/psud]: I think this variable should be renamed either `num_psus` or `psu_count` for clarity.

File: sonic-psud/scripts/psud [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                                stderr=subprocess.STDOUT)
+        stdout = proc.communicate()[0]
+        proc.wait()
+        hwsku = stdout.rstrip('\n')
+    except OSError, e:
+        raise OSError("Cannot detect platform")
+
+    return (platform, hwsku)
+
+
```
Comment (liat-grozovik) [inline sonic-psud/scripts/psud]: Remove extra line

File: sonic-psud/scripts/psud [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+# Loads platform specific psuutil module from source
+def load_platform_psuutil():
```
Comment (liat-grozovik) [inline sonic-psud/scripts/psud]: I believe load platform will be the same for each daemon and while using the defines such as PLATFORM_SPECIFIC_MODULE_NAME and PLATFORM_SPECIFIC_CLASS_NAME can make it a generic utility for each daemon in pmon. What do u think?

File: sonic-psud/scripts/psud [hunk 5]
```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python2
```
[the remark this answers is not in the document]
Comment (kevinwangsk) [reply sonic-psud/scripts/psud]: Just follow the convention of other daemons

File: sonic-psud/scripts/psud [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+
+REDIS_HOSTNAME = "localhost"
+REDIS_PORT = 6379
+REDIS_TIMEOUT_MSECS = 0
+
+PSU_INFO_UPDATE_PERIOD_SECS = 3
+
+#========================== Syslog wrappers ==========================
+
+def log_info(msg, also_print_to_console=False):
```
[the remark this answers is not in the document]
Comment (kevinwangsk) [reply sonic-psud/scripts/psud]: Yes, it is better to have a common file for the generic function of Pmon daemons. But I would suggest it should not in this commit as it is to implement PSU daemon not to improve the generic functions.

File: sonic-psud/scripts/psud (hunk 4 above)
Comment (kevinwangsk) [reply sonic-psud/scripts/psud]: Yes, same with the above comments. Should be in a separate commit to abstract them, not in this commit.

**Pull request merged**